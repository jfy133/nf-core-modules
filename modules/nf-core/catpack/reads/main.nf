process CATPACK_READS {
    tag "${meta.id}"
    label 'process_medium'

    conda "${moduleDir}/environment.yml"
    container "${workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container
        ? 'https://depot.galaxyproject.org/singularity/cat:6.0.1--hdfd78af_0'
        : 'biocontainers/cat:6.0.1--hdfd78af_0'}"

    input:
    tuple val(meta), path(reads)
    tuple val(meta2), path(contigs)
    tuple val(meta3), path(database)
    tuple val(meta4), path(taxonomy)
    tuple val(meta5), path(bins, stageAs: 'bins/')
    val mode
    tuple val(meta6), path(bam_aligned)
    tuple val(meta7), path(bam_unaligned)
    tuple val(meta8), path(contig2classification)
    tuple val(meta9), path(bin2classification)
    tuple val(meta10), path(unclassified2classification)
    tuple val(meta11), path(proteins)
    tuple val(meta12), path(diamond_alignment)

    output:
    tuple val(meta), path("*.txt"), emit: txt
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def insert_reads = meta.single_end ? "--read_file1 ${reads}" : "--read_file1 ${reads[0]} --read_file2 ${reads[1]}"
    def insert_bins = bins ? "--bin_fasta bins/" : ''
    def insert_bam_aligned = bam_aligned ? "-bam1 ${bam_aligned}" : ''
    def insert_bam_unaligned = bam_unaligned ? "--alignment_unmapped ${bam_unaligned}" : ''
    def insert_c2c = contig2classification ? "--c2c ${contig2classification}" : ''
    def insert_b2c = bin2classification ? "--b2c ${bin2classification}" : ''
    def insert_u2c = unclassified2classification ? "--u2c ${unclassified2classification}" : ''
    def insert_proteins = proteins ? "--proteins_fasta ${proteins}" : ''
    def insert_diamond_alignment = diamond_alignment ? "--diamond_alignment ${diamond_alignment}" : ''
    """
    CAT_pack reads \\
        ${insert_reads} \\
        ${args} \\
        -d ${database} \\
        -c ${contigs} \\
        -t ${taxonomy} \\
        -m ${mode} \\
        -o ${prefix}.txt \\
        ${insert_bins} \\
        ${insert_bam_aligned} \\
        ${insert_bam_unaligned} \\
        ${insert_c2c} \\
        ${insert_b2c} \\
        ${insert_u2c} \\
        ${insert_proteins} \\
        ${insert_diamond_alignment}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        catpack: \$(CAT_pack --version | sed 's/CAT_pack pack v//g;s/ .*//g')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def insert_reads = meta.single_end ? "--read_file1 ${reads}" : "--read_file1 ${reads[0]} --read_file2 ${reads[1]}"
    def insert_bins = bins ? "-b-bin_fasta ${bins}" : ''
    def insert_bam_aligned = bam_aligned ? "-bam1 ${bam_aligned}" : ''
    def insert_bam_unaligned = bam_unaligned ? "--alignment_unmapped ${bam_unaligned}" : ''
    def insert_c2c = contig2classification ? "--c2c ${contig2classification}" : ''
    def insert_b2c = bin2classification ? "--b2c ${bin2classification}" : ''
    def insert_u2c = unclassified2classification ? "--u2c ${unclassified2classification}" : ''
    def insert_proteins = proteins ? "--proteins_fasta ${proteins}" : ''
    def insert_diamond_alignment = diamond_alignment ? "--diamond_alignment ${diamond_alignment}" : ''
    """
    echo "    CAT_pack reads \\
        ${insert_reads} \\
        ${args} \\
        -d ${database} \\
        -c ${contigs} \\
        -t ${taxonomy} \\
        -m ${mode} \\
        -o ${prefix}.txt \\
        ${insert_bins} \\
        ${insert_bam_aligned} \\
        ${insert_bam_unaligned} \\
        ${insert_c2c} \\
        ${insert_b2c} \\
        ${insert_u2c} \\
        ${insert_proteins} \\
        ${insert_diamond_alignment}"

    touch ${prefix}.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        catpack: \$(CAT_pack --version | sed 's/CAT_pack pack v//g;s/ .*//g')
    END_VERSIONS
    """
}
