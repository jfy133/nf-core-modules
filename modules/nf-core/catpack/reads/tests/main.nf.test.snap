{
    "sarscov2 - mr": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "txt": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-01-24T13:14:06.906843427"
    },
    "sarscov2 - mcr - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,2978a745fab87a0f0d4e9c3be1004a12"
                ],
                "txt": [
                    [
                        {
                            "id": "test"
                        },
                        "test.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,2978a745fab87a0f0d4e9c3be1004a12"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-01-24T13:23:32.68764654"
    },
    "sarscov2 - cr": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "txt": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-01-24T13:14:01.095037108"
    },
    "sarscov2 - mcr": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "txt": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-01-24T13:13:54.920853143"
    }
}